### ISSUE TITLE
[2.6.0] AkkaModelConversion#parseContentType cannot handle multipart contenttypes

### ISSUE THREAD

**Issue by el-dom** [playframework/playframework#7495]
### Play Version (2.5.x / etc)
2.6.0

### API (Scala / Java / Neither / Both)
Scala

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)
Arch Linux

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)

AkkaModelConversion breaks in 2.6 when using multipart mimetypes. Creating a server which provides content of such a mimetype shows the problem

```
"Foo" should {
   "send multipart content" in {
      import scala.concurrent.duration._
 
      val sccs = Helpers.stubControllerComponents()

      Server.withRouter() {
        case GET(p"/test") =>
          sccs.actionBuilder {
            Ok("data")
              .as("""multipart/mixed; boundary="gc0p4Jq0M:2Yt08jU534c0p"""")
              .withHeaders(LAST_MODIFIED -> "Wed, 10 May 2017 15:00:07 GMT")
          }
      } { implicit port =>
        WsTestClient.withClient { client =>
          Await.result(client.url(s"http://localhost:$port/test").get(), 5.seconds)
        }
      }
    }
  }
```
 
As soon as the GET request gets issued, the following exception occurs
```
java.lang.IllegalArgumentException: requirement failed: Cannot create a MediaType.Multipart here, use `customMultipart` instead!
	at scala.Predef$.require(Predef.scala:224)
	at akka.http.scaladsl.model.MediaType$.customBinary(MediaType.scala:119)
	at akka.http.scaladsl.model.MediaType$.custom(MediaType.scala:176)
	at play.core.server.akkahttp.AkkaModelConversion$$anonfun$parseContentType$2.apply(AkkaModelConversion.scala:234)
	at play.core.server.akkahttp.AkkaModelConversion$$anonfun$parseContentType$2.apply(AkkaModelConversion.scala:233)
	at scala.Option.fold(Option.scala:158)
	at play.core.server.akkahttp.AkkaModelConversion.parseContentType(AkkaModelConversion.scala:233)
	at play.core.server.akkahttp.AkkaModelConversion.convertResultBody(AkkaModelConversion.scala:246)
	at play.core.server.akkahttp.AkkaModelConversion$$anonfun$convertResult$1$$anonfun$apply$1.apply(AkkaModelConversion.scala:201)
	at play.core.server.akkahttp.AkkaModelConversion$$anonfun$convertResult$1$$anonfun$apply$1.apply(AkkaModelConversion.scala:199)
	at akka.http.scaladsl.util.FastFuture$$anonfun$map$extension$1.apply(FastFuture.scala:23)
	at akka.http.scaladsl.util.FastFuture$$anonfun$map$extension$1.apply(FastFuture.scala:23)
	at akka.http.scaladsl.util.FastFuture$.akka$http$scaladsl$util$FastFuture$$strictTransform$1(FastFuture.scala:41)
	at akka.http.scaladsl.util.FastFuture$.transformWith$extension1(FastFuture.scala:55)
	at akka.http.scaladsl.util.FastFuture$.map$extension(FastFuture.scala:23)
	at play.core.server.akkahttp.AkkaModelConversion$$anonfun$convertResult$1.apply(AkkaModelConversion.scala:199)
	at play.core.server.akkahttp.AkkaModelConversion$$anonfun$convertResult$1.apply(AkkaModelConversion.scala:196)
```

Problem here is the contenttype `multipart/mixed`. I think it's caused by the introduction of the `parseContentType` method [here](https://github.com/playframework/playframework/commit/04f7a69762a5b44c67b576d7f99b8a2b0e8a8212#diff-6f79cce2afbe66ce92c239ee90a8d67fR176)

It just calls `MediaType.custom` but it's actually necessary to call `MediaType.customMultipart` for the contenttype at hand.

**Comment by richdougherty** [comment]
Thanks for the report. We'll look into this.

**Comment by richdougherty** [comment]
There's now a fix for this: #7498.

**Comment by zvuki** [comment]
is it related to https://github.com/akka/akka-http/issues/1222

**Comment by richdougherty** [comment]
@zvuki I don't think it's related.

**Comment by richdougherty** [comment]
Hi @el-dom, this issue is now addressed in Play 2.6.1 if you want to try it out.

**Comment by el-dom** [comment]
I hereby confirm, that this is fixed in 2.6.1. Thank you very much.

**Comment by richdougherty** [comment]
Thanks for checking and glad to hear it's working. :)
